### PR TITLE
feat: configure limit type

### PR DESCRIPTION
**Pull request by dimbdr**
Add a limit type combo box, with 2 options:
- Avg -> default type, which works as usual
- Max -> aggregates data focusing on the max value of each axis

Updated documentation

### CODE REVIEW

**Comment by vincentbernat** [comment]
I don't know exactly what you would expect with MAX. As flows are aggregated more and more in the various tables, MAX can be widely different while AVG will stay the same. I suppose you implemented it for a reason, but I fear this would be quite disturbing for many users due to the random side of its effect.

**Comment by moutyque** [comment]
Hello @vincentbernat,

By "MAX can be widely different" do you meant after the aggregation of flows for long terme query ?
If so can we imagine to restrict this MAX for recent flows only ? Like checking against which table we do the query and if the table is not "flows" prompt an error message ?
Thx

**Comment by vincentbernat** [comment]
Yes, there is the aggregation done inside ClickHouse, but also the aggregation done by Netflow/IPFIX. Moreover, I don't understand why you group by the timefilter. What would be the effect of such a thing? Why not just take the max instead of the sum?

I understand your use case, but I need to have a balance between features and maintainability. This code adds a lot of branching. I'll have to maintain that forever.

**Comment by dimbdr** [comment]
I slightly reworked the query, the group by timefilter was indeed unnecessary.
Seems to be more accurate now.

As for the code,I also cleaned up since the query is a bit more easier to write.

File: console/config.go [hunk 1]
```diff
@@ -45,6 +45,8 @@ type VisualizeOptionsConfiguration struct {
 	Dimensions []query.Column `json:"dimensions"`
 	// Limit is the default limit to use
 	Limit int `json:"limit" validate:"min=5"`
+	// LimitType is the default limitType to use
+	LimitType string `json:"limitType"`
```
Comment (vincentbernat) [inline console/config.go:49]: Add validation: `validate:"oneof=avg max"`.

File: console/graph.go [hunk 2]
```diff
@@ -18,8 +18,9 @@ type graphCommonHandlerInput struct {
 	schema         *schema.Component
 	Start          time.Time      `json:"start" binding:"required"`
 	End            time.Time      `json:"end" binding:"required,gtfield=Start"`
-	Dimensions     []query.Column `json:"dimensions"`                          // group by ...
-	Limit          int            `json:"limit" binding:"min=1"`               // limit product of dimensions
+	Dimensions     []query.Column `json:"dimensions"`            // group by ...
+	Limit          int            `json:"limit" binding:"min=1"` // limit product of dimensions
+	LimitType      string         `json:"limitType"`
```
Comment (vincentbernat) [inline console/graph.go:23]: Also validation here.

File: console/line.go [hunk 3]
```diff
@@ -339,7 +334,11 @@ func (c *Component) graphLineHandlerFunc(gc *gin.Context) {
 			if rows[axis][jKey][0] == "Other" {
 				return true
 			}
-			return sums[axis][iKey] > sums[axis][jKey]
+			if input.LimitType == "Max" {
+				return getMax(points[axis][iKey]) > getMax(points[axis][jKey])
+			} else {
+				return sums[axis][iKey] > sums[axis][jKey]
+			}
```
Comment (vincentbernat) [inline console/line.go:341]: Build `maxes` like `sums` is built (or the other way around, also use a function to get the sum).

File: console/line.go [hunk 4]
```diff
@@ -423,6 +422,16 @@ func (c *Component) graphLineHandlerFunc(gc *gin.Context) {
...
 
+func getMax(values []int) int {
+	valMax := values[0]
+	for _, v := range values {
+		if v > valMax {
+			valMax = v
+		}
+	}
+	return valMax
+}
```
Comment (vincentbernat) [inline console/line.go:433]: There is also `slices.Max()`.

**Review by vincentbernat — COMMENTED** [review]
Few minor changes.

File: console/line.go (hunk 4 above)
Comment (dimbdr) [reply console/line.go]: Removed the `getMax` function, no longer useful

File: console/line.go (hunk 3 above)
Comment (dimbdr) [reply console/line.go:345]: Done, `maxes` is constructed the same way `sums` is

File: console/graph.go (hunk 2 above)
Comment (dimbdr) [reply console/graph.go]: Done

File: console/config.go (hunk 1 above)
Comment (dimbdr) [reply console/config.go]: Done

**Comment by vincentbernat** [comment]
Thanks!

**Pull request merged**